Apps/AdminWebClient/src/ClientApp/src/services/restCsvExportService.ts

```diff
@@ -0,0 +1,19 @@
+import { injectable } from "inversify";
+import { ICsvExportService } from "@/services/interfaces";
+
+@injectable()
+export class RestCsvExportService implements ICsvExportService {
+    private readonly BASE_URI: string = "v1/api/CsvExport";
+    public downloadUserInfoCSV(): void {
+        this.downloadCsvExport("GetUserProfiles");
+    }
+    public downloadUserCommentsCSV(): void {
+        this.downloadCsvExport("GetComments");
+    }
+    public downloadUserNotesCSV(): void {
+        this.downloadCsvExport("GetNotes");
+    }
+    private downloadCsvExport(routeName: string) {
+        window.open(`${this.BASE_URI}/${routeName}`);
```

Comment:
Since this is a service it should not interact with the user interface directly. As it is this is breaking the separation of concerns between vue views and supporting code.
Instead how about opening it directly on the vue side?

Reply:
good point, btw, doing it on vue side is also not an ideal solution as vue should not know anything about controller right :-)
How about returning the download url in the service and open it on the vue side?

Reply:
I think is fine if we just put a link there, the path for the csv can be retrieved from the configuration if needed.

Reply:
sounds good!